tests/PPC/TestLifting.cpp

```diff
@@ -0,0 +1,800 @@
+                                 {"r3", uint64_t(0xdd)},
+                                 {"cr0", uint8_t(0)},
+                                 {"pc", uint64_t(0x12)}},
+                                {{"r5", uint64_t(0x1a9)},
+                                 {"cr0", uint8_t(0b100)},
+                                 {"pc", uint64_t(0x16)}},
+                                reg_to_accessor);
+#if LLVM_VERSION_NUMBER < LLVM_VERSION(15, 0)
+  curr_context.enableOpaquePointers();
+#endif
+  TestSpecRunner<PPCState> runner(curr_context, remill::kArchPPC);
+  runner.RunTestSpec(spec);
+}
+
+// Add two registers and set overflow
+TEST(PPCVLELifts, PPCVLEAddOverflow) {
+  llvm::LLVMContext curr_context;
+  // addo r5, r4, r3
+  std::string insn_data("\x7C\xA4\x1E\x14", 4);
+  TestOutputSpec<PPCState> spec(0x12, insn_data,
+                                remill::Instruction::Category::kCategoryNormal,
+                                {{"r4", uint64_t(5000000000000000000)},
+                                 {"r3", uint64_t(5000000000000000000)},
+                                 {"xer_ov", uint8_t(0x0)},
+                                 {"xer_so", uint8_t(0x0)},
+                                 {"pc", uint64_t(0x12)}},
+                                {{"r5", uint64_t(0x8ac7230489e80000)},
+                                 {"xer_ov", uint8_t(0x1)},
+                                 {"xer_so", uint8_t(0x1)},
+                                 {"pc", uint64_t(0x16)}},
+                                reg_to_accessor);
+#if LLVM_VERSION_NUMBER < LLVM_VERSION(15, 0)
+  curr_context.enableOpaquePointers();
+#endif
+  TestSpecRunner<PPCState> runner(curr_context, remill::kArchPPC);
+  runner.RunTestSpec(spec);
+}
+
+// VLE short Branch to Link Register
+TEST(PPCVLELifts, PPCVLEBranchLinkRegister) {
+  llvm::LLVMContext curr_context;
+  // se_blr
+  std::string insn_data("\x00\x04", 2);
+  TestOutputSpec<PPCState> spec(
+      0x12, insn_data, remill::Instruction::Category::kCategoryFunctionReturn,
+      {{"lr", uint64_t(0x4)}, {"pc", uint64_t(0x12)}},
+      {{"lr", uint64_t(0x4)}, {"pc", uint64_t(0x4)}}, reg_to_accessor);
+
+#if LLVM_VERSION_NUMBER < LLVM_VERSION(15, 0)
+  curr_context.enableOpaquePointers();
+#endif
+  TestSpecRunner<PPCState> runner(curr_context, remill::kArchPPC);
+  runner.RunTestSpec(spec);
+}
+
+// VLE short Branch to Link Register and Link
+TEST(PPCVLELifts, PPCVLEBranchLinkRegisterAndLink) {
+  llvm::LLVMContext curr_context;
+  // se_blrl
+  std::string insn_data("\x00\x05", 2);
+  TestOutputSpec<PPCState> spec(
+      0x12, insn_data, remill::Instruction::Category::kCategoryFunctionReturn,
+      {{"lr", uint64_t(0x4)}, {"pc", uint64_t(0x12)}},
+      {{"lr", uint64_t(0x14)}, {"pc", uint64_t(0x4)}}, reg_to_accessor);
+
+#if LLVM_VERSION_NUMBER < LLVM_VERSION(15, 0)
+  curr_context.enableOpaquePointers();
+#endif
+  TestSpecRunner<PPCState> runner(curr_context, remill::kArchPPC);
+  runner.RunTestSpec(spec);
```

Comment:
I see this copy & pasted all over the place -> should be present as little times as possible.